tcpMaxConn.go

```diff
@@ -24,11 +27,19 @@ func main() {
 		debugOut = os.Stderr
 	}
 
-	connStatusCh := mtcpclient.StartReportingLogic(*numberConnectionsPtr, *reportingIntervalPtr)
-
-	mtcpclient.MultiTCPConnect(*numberConnectionsPtr, *delayPtr, *hostPtr, *portPtr, connStatusCh, debugOut)
+	connStatusCh, connStatusTracker := mtcpclient.StartBackgroundReporting(*numberConnectionsPtr, *reportingIntervalPtr)
+	closureCh := mtcpclient.StartBackgroundClosureTrigger(connStatusTracker)
+	mtcpclient.MultiTCPConnect(*numberConnectionsPtr, *delayPtr, *hostPtr, *portPtr, connStatusCh, closureCh, debugOut)
 
+	printClosureReport(*hostPtr, *portPtr, connStatusTracker)
 	fmt.Fprintln(debugOut, "\nTerminating Program")
 }
+func printClosureReport(host string, port int, connections []tcpclient.Connection) {
+	// workaround to allow last status updates to be collected properly
+	time.Sleep(time.Duration(50) * time.Millisecond)
```

Comment:
Even it's a workaround, a properly named constant is always useful to understand, something like `const timeToWaitForClosureReport`

Reply:
good point